src/read.jl

```diff
@@ -261,3 +255,225 @@ function read_demand_flexibility(filepath, interval)::DemandFlexibility
 
     return demand_flexibility
 end
+
+"""
+    _make_bus_demand_weighting(case)
+
+Given a Case object, build a sparse matrix that indicates the weighting of each bus in 
+    each zone.
+"""
+function _make_bus_demand_weighting(case::Case)::SparseMatrixCSC
+    bus_idx = 1:length(case.busid)
+    bus_df = DataFrames.DataFrame(;
+        name=case.busid, load=case.bus_demand, zone=case.bus_zone
+    )
+    zone_demand = DataFrames.combine(DataFrames.groupby(bus_df, :zone), :load => sum)
+    zone_list = sort(collect(Set(case.bus_zone)))
+    zone_idx = 1:length(zone_list)
+    zone_id2idx = Dict(zone_list .=> zone_idx)
+    bus_df_with_zone_load = DataFrames.innerjoin(bus_df, zone_demand; on=:zone)
+    bus_share = bus_df[:, :load] ./ bus_df_with_zone_load[:, :load_sum]
+    bus_zone_idx = Int64[zone_id2idx[z] for z in case.bus_zone]
+    zone_to_bus_shares = sparse(bus_zone_idx, bus_idx, bus_share)::SparseMatrixCSC
+    return zone_to_bus_shares
+end
+
+"""
+    reformat_demand_flexibility_input(case, demand_flexibility, sets)
+
+Inspect the raw input files of flexiblility amount and cost, convert the zone/bus mixed 
+    data to bus data
+"""
+function reformat_demand_flexibility_input(
+    case::Case, demand_flexibility::DemandFlexibility, sets::Sets
+)
+
+    # check consistency of flexibility input headers
+    if !all(
+        sort(names(demand_flexibility.flex_amt_up)) .==
+        sort(names(demand_flexibility.flex_amt_dn)),
+    )
+        @error("The flexible bus/load zone specified in the up/down input csvs do not 
+               match. Please check the input files to make sure every flexible bus or
+               load zone has corresponding columns in both flexibility csvs.")
+        throw(ErrorException("See above."))
```

Comment:
With @danielolsen's suggested change to `gurobi.jl`, is this method for stopping the simulation and producing an error message still needed? If it works as expected, only `throw(ExceptionError("The error message."))` should be needed, right?